src/api/api.cc

```diff
@@ -11743,6 +11740,22 @@ extern "C" DLLEXPORT void V8RecordReplayOnNavigationEvent(const char* kind, cons
   gRecordReplayOnNavigationEvent(kind, url);
 }
 
+#define MaxJsStackTraceSize (4096)
+static char gLastJsStackTrace[MaxJsStackTraceSize];
+
+extern "C" DLLEXPORT const char* V8RecordReplayGetCurrentJSStackTmp() {
+  CHECK(recordreplay::IsRecordingOrReplaying());
+  std::stringstream stack;
+
+  i::Isolate* isolate = internal::Isolate::TryGetCurrent();
+  i::HandleScope scope(isolate);
+  if (isolate) {
+    isolate->PrintCurrentStackTrace(stack);
+  }
+  strncpy(gLastJsStackTrace, stack.str().c_str(), MaxJsStackTraceSize);
```

Comment:
Calling strncpy like this is broken because strncpy doesn't null terminate the string if the source is too long (strncpy is a terribly designed API).  Why are we jumping through these hoops of returning a pointer to static storage anyways?  It seems better if this function takes a `std::string*` outparam instead (I don't think C-linkage functions can return an std::string).

Reply:
I was afraid of doing stuff with `std::string` because of C-linkage. I'll try the outparam route.